### PR TITLE
Fix `copy-bulk.sh` and optimise dependencies

### PR DESCRIPTION
**Pull request by Silic0nS0ldier**
Fixes shell usage and optimises dependency logic (gains will be more noticeable with https://github.com/bazelbuild/bazel/pull/20434).

### CODE REVIEW

File: internal/npm_install/bulk_copy.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
```
Comment (christianscott) [inline internal/npm_install/bulk_copy.sh:1]: missed this on the first PR. why not
```suggestion
#! /usr/bin/env bash
```
?

**Review by christianscott — CHANGES_REQUESTED** [review]
Why do we need to remove `pipefail`?

**Comment by Silic0nS0ldier** [comment]
> Why do we need to remove pipefail?

Because I wrote the script to use `sh` (POSIX shell) which does not support that option.

The suggestion of outright using `bash` would make it safe though.

**Review by christianscott — APPROVED** [review]
LGTM

**Pull request merged**